reactor-netty-http/src/main/java/reactor/netty/http/client/HttpClientConnect.java

```diff
@@ -209,24 +211,32 @@ public void subscribe(CoreSubscriber<? super Connection> actual) {
 				if (handler.toURI.isSecure()) {
 					if (_config.sslProvider == null) {
 						_config = new HttpClientConfig(config);
-						for (int i = 0; i < _config.protocols.length; i++) {
-							if (_config.protocols[i] == HttpProtocol.H2C) {
-								_config.protocols[i] = HttpProtocol.H2;
+						if ((_config._protocols & HttpClientConfig.h2c) == HttpClientConfig.h2c &&
```

Comment:
consider extracting that choice algorithm into a separate method, which can be tested separately

Reply:
done